src/WooCommerce.php

```diff
@@ -985,6 +985,41 @@ public static function getTaxonomyTermsAsSelectOptions($taxonomy) {
     return is_wp_error($terms) ? [] : $term_options;
   }
 
+  /**
+   * Retrieves the earliest "back in stock" date from a product or its variations.
+   * If there's stock, the product is considered available and nothing is returned.
+   * @param WC_Product
+   *    The product object.
+   * @return string
+   *    The earliest back in stock date, if any.
+   */
+  public static function getEarliestBackInStock($product) {
+    if ($product->get_stock_quantity() > 0) {
+      return '';
+    }
+
+    $meta_key = '_' . Plugin::PREFIX . '_back_in_stock_date';
+    $back_in_stock_date = get_post_meta($product->get_id(), $meta_key, TRUE) ?? '';
+
+    if ($product->is_type('variable')) {
+      $variations = $product->get_available_variations();
+      foreach ($variations as $variation) {
+        $variation_obj = wc_get_product($variation['variation_id']);
```

Comment:
Maybe a more suitable name? 🤔 
```suggestion
        $product_variation = wc_get_product($variation['variation_id']);
```

Reply:
I'll change it, but I named it that way to differenciate it between the "array like" returned by `get_available_variations` as opposed to the Object returned by `wc_get_product`.